velox/exec/Window.cpp

```diff
@@ -284,8 +284,25 @@ void updateKRowsOffsetsColumn(
   // moves ahead.
   int precedingFactor = isKPreceding ? -1 : 1;
   for (auto i = 0; i < numRows; i++) {
-    rawFrameBounds[i] =
-        (startRow + i) + vector_size_t(precedingFactor * offsets[i]);
+    auto startValue = (int64_t)(startRow + i) + precedingFactor * offsets[i];
+    if (startValue < INT32_MIN) {
+      rawFrameBounds[i] = 0;
+    } else if (startValue > INT32_MAX) {
+      // computeValidFrames will replace INT32_MAX set here
+      // with partition's final row index.
+      rawFrameBounds[i] = INT32_MAX;
+    } else {
+      rawFrameBounds[i] = startValue;
+    }
+  }
+}
+
+// Get the start index that integer overflow can happen for kFollowing.
+FOLLY_ALWAYS_INLINE int32_t getOverflowStart(int64_t startValue) {
```

Comment:
Nit : Maybe call this overflowLimit (instead of overflowStart) and the parameter can be just called k.

This function is only used in one place. We can move this logic within the below function.

Reply:
@aditi-pandit, thanks for your suggestion. As you pointed out, this small piece of code is only used in one place. So I just put it into the caller side, not making it a function or lambda.